3ds2/src/main/java/com/adyen/checkout/adyen3ds2/internal/ui/DefaultAdyen3DS2Delegate.kt

```diff
@@ -481,13 +508,36 @@ internal class DefaultAdyen3DS2Delegate(
 
     override fun onCompletion(result: ChallengeResult) {
         when (result) {
```

Comment:
Maybe this `when` can return the subtype and then the track method can be called once